.github/workflows/ci.yml

```diff
@@ -134,7 +132,7 @@ jobs:
       platforms: linux/${{ matrix.platform }}
       dockerfile: build/assets.Dockerfile
       build_target: prereqs
-      toolchain: gcc11
+      toolchain: llvm
```

Comment:
This is the whole purpose of this PR; the installer is now built against libc++ instead of against libstdc++. Docker images and Python wheels are left as is.

Reply:
It wasn't obvious to me. One can use libc++ with either clang++ or g++.

Reply:
Right, but I think it is good to aim for having a toolchain that can build itself, or put differently: If the compiler used to build the shipped CUDA-Q libraries is the same compiler as the one that CUDA-Q uses to build applications/libraries. I don't quite see your point here; at some point in our build (whether that be for the bootstrap, the llvm libs build, or the cudaq applications build) we will start building everything with the clang/llvm toolchain. This build accomplishes this in the least painful way. This way of doing that is fully tested and maintained by llvm. Accomplishing the same with gcc is not, so I am not sure I understand what you hope to gain for building more steps with gcc before ultimately fully relying on clang/llvm?

Reply:
I guess "llvm" is such an overloaded name that it really has lost any meaning. gcc-11, g++-12, clang++-16 ... are names that have an intuitive connection for me. "llvm" is like saying "unknown" or "special sauce". Probably just me.